tools/ec2recovery.py

```diff
@@ -0,0 +1,284 @@
+    return result
+
+
+def select_snapshot(snapshots):
+    while True:
+        clear_screen()
+        result = ["NA"]
+        sel_num = 0
+        print "\n  Snapshot Date Selection\n\n"
+        for snap in snapshots:
+            result.append(snap)
+            sel_num += 1
+            snap_option = "\t%s) %s\t" % (sel_num, snap)
+            for snapshot in snapshots[snap]:
+                snap_option += "[%s | %s | %s]\n\t\t\t" % (
+                    snapshot[0],
+                    snapshot[1],
+                    snapshot[2])
+            print snap_option
+        print "\n\t0) Exit\n\n"
+        ami_num = input("  Restore Date [0-%s]: " % sel_num)
+        if ami_num in range(0, len(result)):
+            return result[ami_num]
+
+
+def select_ami(amis):
+    while True:
+        clear_screen()
+        result = ["NA"]
+        sel_num = 0
+        for ami in amis:
+            sel_num += 1
+            vals = [ami, amis[ami][0], amis[ami][1].strftime("%m-%d-%Y")]
+            result.append(vals[1])
+            line_value = "\t%s) %s %s (%s)" % (
+                sel_num, vals[0],
+                vals[2], vals[1])
+            if sel_num == 1:
+                print "\n  AMI Selection\n\n"
+            print line_value
+        print "\t0) Exit\n\n"
+        ami_num = input("  Desired AMI [0-%s]: " % sel_num)
+        if ami_num in range(0, len(result)):
+            return result[ami_num]
+
+
+def create_bdm(snaps):
+    bdm = []
+    for snap in snaps:
+        volume = {}
+        volume['DeviceName'] = snap[1]
+        volume['Ebs'] = {}
+        volume['Ebs']['SnapshotId'] = snap[2]
+        bdm.append(volume)
+    return bdm
+
+
+def create_tags(tags):
+    result = []
+    for i in range(len(tags)):
+        key = tags[i]['Key']
+        value = tags[i]['Value']
+        if key == 'Name':
+            value = "%s-restored" % value
+        if key == 'RestoreDate' or 'aws:' in key:
+            continue
+        result.append({
+            'Key': key,
+            'Value': value})
+    result.append({
+        'Key': 'RestoreDate',
+        'Value': datetime.now().strftime("%Y-%m-%d %H:%M:%S")})
+    return result
+
+
+def create_instance(session, dry_run, ami_id, instance, bdm, user_data):
+    # Create volumes from snapshots
+    ec2 = session.resource('ec2')
+    instances = ec2.create_instances(
+        DryRun=dry_run,
+        ImageId=ami_id,
+        MinCount=1,
+        MaxCount=1,
+        KeyName=instance['KeyName'],
+        SecurityGroupIds=pull_sg_ids(instance),
+        InstanceType=instance['InstanceType'],
+        SubnetId=instance['SubnetId'],
+        BlockDeviceMappings=bdm,
+        IamInstanceProfile=pull_iam_profile(instance),
+        UserData=user_data)
+    instance_id = instances[0].instance_id
+    restored = ec2.Instance(instance_id)
+    print "\n\tRestoring instance from snapshot..."
+    while restored.state['Name'] != 'running':
+        time.sleep(5)
```

Comment:
there's waiters in the sdk that automate this for  you and also handle failure states (ie instance never gets to running). see some in resources/ebs.py